### PR TITLE
Add translation of the legacy config into default header validator config

### PR DESCRIPTION
**Pull request by yanavlasov**
Additional Description:
This enables the default header validator for all tests that use HCM (i.e. integration tests).

After merging this PR, this change can then be merged into #22537 to verify default HV effect on integration tests.

Risk Level: Low, UHV enabled build only
Testing: Unit Tests
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

Signed-off-by: Yan Avlasov <yavlasov@google.com>

### CODE REVIEW

File: test/extensions/filters/network/http_connection_manager/config_test.cc [hunk 1]
```diff
@@ -2596,6 +2598,43 @@ class TestHeaderValidatorFactoryConfig : public Http::HeaderValidatorFactoryConf
   }
 };
 
+// Override the default config factory for the proto that is expected to be created when
+// header validator config was not specified.
```
Comment (alyssawilk) [inline test/extensions/filters/network/http_connection_manager/config_test.cc:2602]: wait, why is the default factory not the default?

File: source/extensions/filters/network/http_connection_manager/config.cc [hunk 2]
```diff
@@ -133,27 +134,51 @@ createHeaderValidatorFactory([[maybe_unused]] const envoy::extensions::filters::
...
-  if (config.has_typed_header_validation_config()) {
 #ifdef ENVOY_ENABLE_UHV
-    auto* factory = Envoy::Config::Utility::getFactory<Http::HeaderValidatorFactoryConfig>(
-        config.typed_header_validation_config());
-    if (!factory) {
-      throw EnvoyException(fmt::format("Header validator extension not found: '{}'",
-                                       config.typed_header_validation_config().name()));
-    }
+  ::envoy::config::core::v3::TypedExtensionConfig legacy_header_validator_config;
+  if (!config.has_typed_header_validation_config()) {
```
Comment (alyssawilk) [inline source/extensions/filters/network/http_connection_manager/config.cc:139]: comment here as well?
something along the lines of "if using UHV with no configuration, ensure the defaults match Envoy's original behavior"

**Review by alyssawilk — COMMENTED** [review]
sweet, thanks!
/wait

File: source/extensions/filters/network/http_connection_manager/config.cc (hunk 2 above)
Comment (yanavlasov) [reply source/extensions/filters/network/http_connection_manager/config.cc:139]: Done

File: test/extensions/filters/network/http_connection_manager/config_test.cc (hunk 1 above)
Comment (yanavlasov) [reply test/extensions/filters/network/http_connection_manager/config_test.cc]: This override is for the test to see what UHV config proto is created by HCM factory by default.

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**